### PR TITLE
dev-guide changes DONE 

### PR DESCRIPTION
**Pull request by Jaideep1702**
# Pull Request Template

## Description

This pull request introduces the following enhancements to the Keploy developer guide:

1. **Note Added for `--pull always` Flag**: I have added a note under the "Setup Keploy via Docker" section, recommending the use of the `--pull always` flag in the alias when using the deployed Keploy binary image.

2. **Heading Numbering Fixed**: The numbering of the headings in the developer guide has been corrected.

3. provided a description emphasizing the importance of running this command inside the Keploy repository and ensuring that there is no directory named 'keploy' within the main Keploy repository when building the image. 

Thank you for considering these updates.

Fixes # (issue)
[docs]: Minor fixes in dev-guide #947

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue).

## How Has This Been Tested?

I have built the project on my local system to validate these changes. 

## Additional Context (Please include any Screenshots/gifs if relevant)

1. 
image

2.
image

3.
image

...

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [x] I have added corresponding tests.

### CODE REVIEW

File: versioned_docs/version-2.0.0/keploy-explained/dev-guide.md
Comment (404fixer) [inline versioned_docs/version-2.0.0/keploy-explained/dev-guide.md:1]: You have to add the `--pull always` flag in the alias. Don't need to add _Note_ below it.

Comment (404fixer) [inline versioned_docs/version-2.0.0/keploy-explained/dev-guide.md:1]: Don't mention it as a note. Write it below the **Build Docker Image:** subheading. 
And rephrase the sentence to something like: _Run the below command inside the keploy respository and make sure there is no directory by the name of keploy inside the main keploy repository._

Comment (404fixer) [inline versioned_docs/version-2.0.0/keploy-explained/dev-guide.md:1]: Update this command to `docker build -t <nameOfImage> .`

Comment (Jaideep1702) [reply versioned_docs/version-2.0.0/keploy-explained/dev-guide.md:1]: done

Comment (Jaideep1702) [reply versioned_docs/version-2.0.0/keploy-explained/dev-guide.md:1]: done

Comment (Jaideep1702) [reply versioned_docs/version-2.0.0/keploy-explained/dev-guide.md:1]: done

Comment (404fixer) [reply versioned_docs/version-2.0.0/keploy-explained/dev-guide.md:1]: You need to update the alias. It should be `alias keployV2='sudo docker run --pull always --name keploy-ebpf -p 16789:16789 --network keploy-network --privileged --pid=host -it -v "$(pwd)":/files -v /sys/fs/cgroup:/sys/fs/cgroup -v /sys/kernel/debug:/sys/kernel/debug -v /sys/fs/bpf:/sys/fs/bpf -v /var/run/docker.sock:/var/run/docker.sock --rm  <nameOfImage>`. And remove this _Add `--pull always` flag in the alias when using the deployed keploy binary image._

Comment (Jaideep1702) [reply versioned_docs/version-2.0.0/keploy-explained/dev-guide.md:1]: done

Comment (Jaideep1702) [reply versioned_docs/version-2.0.0/keploy-explained/dev-guide.md:1]: done

Comment (404fixer) [reply versioned_docs/version-2.0.0/keploy-explained/dev-guide.md:1]: Sorry, I forgot to mention the image name in the alias. Please make it this: `alias keployV2='sudo docker run --pull always --name keploy-ebpf -p 16789:16789 --network keploy-network --privileged --pid=host -it -v "$(pwd)":/files -v /sys/fs/cgroup:/sys/fs/cgroup -v /sys/kernel/debug:/sys/kernel/debug -v /sys/fs/bpf:/sys/fs/bpf -v /var/run/docker.sock:/var/run/docker.sock --rm  ghcr.io/keploy/keploy`

Comment (Jaideep1702) [reply versioned_docs/version-2.0.0/keploy-explained/dev-guide.md:1]: done

**Comment by Sonichigo** [comment]
Hey @Jaideep1702 !! Can you fix the lint issue ?

**Review by 404fixer — APPROVED** [review]
LGTM

**Pull request merged**